### PR TITLE
Addresses issues 147-166 

### LINKED ISSUE #147: NTR: live fences
Term: live fences
Definition: Fences established by planting large cuttings, that easily produce roots and on which several strings of wire are attached with the obvious purpose of keeping livestock in or out
Source: Harvey, C.A., Villanueva, C., Villacís, J., Chacón, M., Muñoz, D., López, M., Ibrahim, M., Gómez, R., Taylor, R., Martinez, J. and Navas, A., 2005. Contribution of live fences to the ecological integrity of agricultural landscapes. Agriculture, ecosystems & environment, 111(1-4), pp.200-230.
ecological integrity of agricultural
landscapes
Parent: AGRO:00000516
Related: AGRO:00000584, trees and crops, trees on crop land , zoned, on boundaries, live fences
ORCID: https://orcid.org/0000-0003-3985-4911

### LINKED ISSUE #148: NTR: boundary planting
Term: boundary planting 
Definition: 
Source: Sinclair, F.L., 1999. A general classification of agroforestry practice. Agroforestry systems, 46(2), pp.161-180.
Parent: AGRO:00000516
Related: AGRO:00000584, trees and crops, trees on crop land , zoned, on boundaries, boundary planting
ORCID: https://orcid.org/0000-0003-3985-4911

### LINKED ISSUE #149: NTR: windbreaks/timber belts
Term: windbreaks/timber belts 
Definition: Windbreaks are linear plantings of trees and shrubs designed to enhance crop production, protect people and livestock, and benefit soil and water conservation.
Source: GACS: http://id.agrisemantics.org/gacs/C12778
Parent: AGRO:00000516
Related: AGRO:00000584, trees and crops, trees on crop land , zoned, on boundaries, windbreaks/timber belts
ORCID: https://orcid.org/0000-0003-3985-4911

### LINKED ISSUE #152: NTR: production hedges
Term: production hedges 
Definition: 
Source: Sinclair, F.L., 1999. A general classification of agroforestry practice. Agroforestry systems, 46(2), pp.161-180.
Parent: AGRO:00000516
Related: AGRO:00000584, trees and crops, trees on crop land , zoned, in rows, production hedges
ORCID: https://orcid.org/0000-0003-3985-4911

### LINKED ISSUE #153: NTR: fodder banks
Term: fodder banks 
Definition: 
Source: Sinclair, F.L., 1999. A general classification of agroforestry practice. Agroforestry systems, 46(2), pp.161-180.
Parent: AGRO:00000516
Related: AGRO:00000584, trees and crops, trees on crop land , zoned, in blocks, fodder banks
ORCID: https://orcid.org/0000-0003-3985-4911

### LINKED ISSUE #155: NTR: farm woodlands
Term: farm woodlands 
Definition: 
Source: Sinclair, F.L., 1999. A general classification of agroforestry practice. Agroforestry systems, 46(2), pp.161-180.
Parent: AGRO:00000516
Related: AGRO:00000584, trees and crops, trees on crop land , zoned, in blocks, farm woodlands
ORCID: https://orcid.org/0000-0003-3985-4911

### LINKED ISSUE #156: NTR: regular
Term: regular 
Definition: 
Source: Prevedello, J.A., Almeida‐Gomes, M. and Lindenmayer, D.B., 2018. The importance of scattered trees for biodiversity conservation: A global meta‐analysis. Journal of Applied Ecology, 55(1), pp.205-214.
Parent: AGRO:00000516
Related: AGRO:00000584, trees and crops, trees on crop land , dispersed (trees in fields/on range), reguals,
ORCID: https://orcid.org/0000-0003-3985-4911

### LINKED ISSUE #158: NTR: cropping phase of taungya
Term: cropping phase of taungya
Definition: 
Source: Sinclair, F.L., 1999. A general classification of agroforestry practice. Agroforestry systems, 46(2), pp.161-180.
Parent: AGRO:00000516
Related: AGRO:00000584, trees and crops, crops on tree-land, plantation forest/trees, cropping phase of taungya,
ORCID: https://orcid.org/0000-0003-3985-4911

### LINKED ISSUE #160: NTR: forest gardens
Term: forest gardens
Definition: 
Source: Sinclair, F.L., 1999. A general classification of agroforestry practice. Agroforestry systems, 46(2), pp.161-180.
Parent: AGRO:00000516
Related: AGRO:00000584, trees and crops, crops on tree-land, natural forest/tree cover, forest gardens,
ORCID: https://orcid.org/0000-0003-3985-4911

### LINKED ISSUE #161: NTR: cropping phase of shifting cultivation
Term: cropping phase of shifting cultivation
Definition: 
Source: Sinclair, F.L., 1999. A general classification of agroforestry practice. Agroforestry systems, 46(2), pp.161-180.
Parent: AGRO:00000516
Related: AGRO:00000584, trees and crops, crops on tree-land, natural forest/tree cover, cropping phase of shifting cultivation,
ORCID: https://orcid.org/0000-0003-3985-4911

### LINKED ISSUE #162: NTR: forest/plantation grazing
Term: forest/plantation grazing 
Definition: 
Source: Sinclair, F.L., 1999. A general classification of agroforestry practice. Agroforestry systems, 46(2), pp.161-180.
Parent: AGRO:00000516
Related: AGRO:00000584, trees and crops, crops on tree-land, natural forest/tree cover, forest/plantation grazing ,
ORCID: https://orcid.org/0000-0003-3985-4911

### LINKED ISSUE #164: as barrier hedges
Term: 
Definition: 
Source: 
Parent: 
Related: AGRO:00000516, AGRO:00000584, trees and crops, trees on crop land , zoned, on contours, as barrier hedges
ORCID: https://orcid.org/0000-0003-3985-4911

### LINKED ISSUE #165: alley cropping
Term: 
Definition: A cropping system that involves growing crops in a wide strip, typically 6 meters in width, between lines of closely planted, fast-growing trees or shrubs. These woody species are usually leguminous and are pruned frequently to provide a mulching material and nutrients to the crop in the alley.
Source: 
Parent: 
Related: AGRO:00000516, AGRO:00000584, trees and crops, trees on crop land , zoned, in rows, alley cropping 
ORCID: https://orcid.org/0000-0003-3985-4911

### LINKED ISSUE #166: improved fallow
Term: 
Definition: 
Source: 
Parent: 
Related: AGRO:00000516, AGRO:00000584, forests/tress and people , (sustainable) multipurpose utlization of natural forest/tree cover , fallow phase of shifting cultivation , improved fallow, 
ORCID: https://orcid.org/0000-0003-3985-4911

### PR DESCRIPTION
**Pull request by KrishnaTO**
See individual commits

closes #147
closes #148
closes #149
closes #152
closes #153
closes #155
closes #156
closes #158
closes #160
closes #161
closes #162
closes #164
closes #165
closes #166

### CODE REVIEW

File: src/ontology/agro-edit.owl [hunk 1]
```diff
@@ -4679,6 +4692,8 @@ AnnotationAssertion(Annotation(<http://purl.org/dc/elements/1.1/source> "https:/
 AnnotationAssertion(<http://purl.org/dc/elements/1.1/creator> <http://purl.obolibrary.org/obo/AGRO_00000579> <https://orcid.org/0000-0003-3985-4911>)
 AnnotationAssertion(<http://www.geneontology.org/formats/oboInOwl#hasExactSynonym> <http://purl.obolibrary.org/obo/AGRO_00000579> "alleycropping"@en)
 AnnotationAssertion(<http://www.geneontology.org/formats/oboInOwl#hasExactSynonym> <http://purl.obolibrary.org/obo/AGRO_00000579> "hedgerow intercropping"@en)
+AnnotationAssertion(rdfs:comment <http://purl.obolibrary.org/obo/AGRO_00000579> "A cropping system that involves growing crops in a wide strip, typically 6 meters in width, between lines of closely planted, fast-growing trees or shrubs."@en)
```
Comment (celineaubert) [inline src/ontology/agro-edit.owl:4730]: where does this comment come from?

File: src/ontology/agro-edit.owl [hunk 2]
```diff
@@ -5446,6 +5462,127 @@ AnnotationAssertion(rdfs:label <http://purl.obolibrary.org/obo/AGRO_00010155> "i
...
+
+# Class: <http://purl.obolibrary.org/obo/AGRO_00010157> (boundary planting)
+
+AnnotationAssertion(Annotation(<http://www.geneontology.org/formats/oboInOwl#hasDbXref> "Technical Specification Boundary Planting. https://takingroot.org/wp-content/uploads/2015/04/takingroot-boundary-planting.pdf") <http://purl.obolibrary.org/obo/IAO_0000115> <http://purl.obolibrary.org/obo/AGRO_00010157> "A planting process where trees or hedges are planted along boundaries to delimitate property, agricultural fields, pastures, roads or any other place where fences might be located."@en)
+AnnotationAssertion(<http://purl.org/dc/elements/1.1/creator> <http://purl.obolibrary.org/obo/AGRO_00010157> "https://orcid.org/0000-0003-3985-4911")
+AnnotationAssertion(<http://purl.org/dc/elements/1.1/date> <http://purl.obolibrary.org/obo/AGRO_00010157> "2021-12-27T16:54:13Z"^^xsd:dateTime)
+AnnotationAssertion(rdfs:label <http://purl.obolibrary.org/obo/AGRO_00010157> "boundary planting"@en)
+SubClassOf(<http://purl.obolibrary.org/obo/AGRO_00010157> <http://purl.obolibrary.org/obo/AGRO_00000231>)
```
Comment (celineaubert) [inline src/ontology/agro-edit.owl:5696]: I wonder if this term should go under ''live fence'' class. I have found that exact syn are:  barrier planting, Trees on boundaries and live fence. 
The term is a bit tricky as it includs a verb. But I think the term does not describe a process but the finality of this process (based on the other term in ERA list), and is therefore a continuant.  

Definition: Live fence made of trees. Boundary planting are used to delimitate property, agricultural fields, pastures, roads or any other place where fences might be located.

Source: Baker, K., Baumann, D., Michaud, J.S., van Mossel-Forrester, B., 2014. Technical Specification Boundary Planting. Registered Plan Vivo Project: The CommuniTree Carbon Project. 64p. URL: https://takingroot.org/wp-content/uploads/2015/04/takingroot-boundary-planting.pdf

File: src/ontology/agro-edit.owl [hunk 3]
```diff
@@ -5446,6 +5462,127 @@ AnnotationAssertion(rdfs:label <http://purl.obolibrary.org/obo/AGRO_00010155> "i
...
+
+# Class: <http://purl.obolibrary.org/obo/AGRO_00010158> (windbreak)
+
+AnnotationAssertion(Annotation(<http://www.geneontology.org/formats/oboInOwl#hasDbXref> "https://www.fs.usda.gov/nac/practices/windbreaks.php") <http://purl.obolibrary.org/obo/IAO_0000115> <http://purl.obolibrary.org/obo/AGRO_00010158> "Windbreaks are linear plantings of trees and shrubs designed to enhance crop production, protect people and livestock, and benefit soil and water conservation."@en)
+AnnotationAssertion(<http://purl.org/dc/elements/1.1/creator> <http://purl.obolibrary.org/obo/AGRO_00010158> <https://orcid.org/0000-0003-3985-4911>)
+AnnotationAssertion(<http://purl.org/dc/elements/1.1/date> <http://purl.obolibrary.org/obo/AGRO_00010158> "2022-01-03T23:59:15Z"^^xsd:dateTime)
+AnnotationAssertion(<http://www.geneontology.org/formats/oboInOwl#hasExactSynonym> <http://purl.obolibrary.org/obo/AGRO_00010158> "shelterbelt")
+AnnotationAssertion(rdfs:label <http://purl.obolibrary.org/obo/AGRO_00010158> "windbreak"@en)
+SubClassOf(<http://purl.obolibrary.org/obo/AGRO_00010158> <http://purl.obolibrary.org/obo/AGRO_00000584>)
```
Comment (celineaubert) [inline src/ontology/agro-edit.owl:5705]: I wonder if the subcass: http://purl.obolibrary.org/obo/ENVO_01000408 would be better? Again, this is not a process but the result of the process of planting trees.

File: src/ontology/agro-edit.owl (hunk 1 above)
Comment (KrishnaTO) [reply src/ontology/agro-edit.owl:4730]: From #165 , where a revised definition was proposed, it was added as a comment due to the indefinite but typical content.

File: src/ontology/agro-edit.owl [hunk 4]
```diff
@@ -5446,6 +5462,127 @@ AnnotationAssertion(rdfs:label <http://purl.obolibrary.org/obo/AGRO_00010155> "i
...
+SubClassOf(<http://purl.obolibrary.org/obo/AGRO_00010160> <http://purl.obolibrary.org/obo/AGRO_00000584>)
+SubClassOf(<http://purl.obolibrary.org/obo/AGRO_00010160> ObjectSomeValuesFrom(<http://purl.obolibrary.org/obo/RO_0000052> <http://purl.obolibrary.org/obo/AGRO_00000231>))
+
+# Class: <http://purl.obolibrary.org/obo/AGRO_00010161> (hedgerow trees)
+
+AnnotationAssertion(Annotation(<http://www.geneontology.org/formats/oboInOwl#hasDbXref> "https://en.wikipedia.org/w/index.php?title=Hedge&oldid=1062906324") <http://purl.obolibrary.org/obo/IAO_0000115> <http://purl.obolibrary.org/obo/AGRO_00010161> "Tree plantings which are grown in hedgerows but have been allowed to reach their full height and width."@en)
+AnnotationAssertion(<http://purl.org/dc/elements/1.1/creator> <http://purl.obolibrary.org/obo/AGRO_00010161> <https://orcid.org/0000-0003-3985-4911>)
+AnnotationAssertion(<http://purl.org/dc/elements/1.1/date> <http://purl.obolibrary.org/obo/AGRO_00010161> "2022-01-04T00:23:05Z"^^xsd:dateTime)
+AnnotationAssertion(<http://www.geneontology.org/formats/oboInOwl#hasRelatedSynonym> <http://purl.obolibrary.org/obo/AGRO_00010161> "production hedges")
```
Comment (celineaubert) [inline src/ontology/agro-edit.owl:5731]: I have asked Peter the definition of this term that does not appear in the source he mentionned neither on any other document.

File: src/ontology/agro-edit.owl [hunk 5]
```diff
@@ -5446,6 +5462,127 @@ AnnotationAssertion(rdfs:label <http://purl.obolibrary.org/obo/AGRO_00010155> "i
...
+AnnotationAssertion(<http://purl.org/dc/elements/1.1/creator> <http://purl.obolibrary.org/obo/AGRO_00010166> <https://orcid.org/0000-0003-3985-4911>)
+AnnotationAssertion(<http://purl.org/dc/elements/1.1/date> <http://purl.obolibrary.org/obo/AGRO_00010166> "2022-01-04T01:00:02Z"^^xsd:dateTime)
+AnnotationAssertion(<http://www.geneontology.org/formats/oboInOwl#hasBroadSynonym> <http://purl.obolibrary.org/obo/AGRO_00010166> "Woodland garden"@en)
+AnnotationAssertion(rdfs:label <http://purl.obolibrary.org/obo/AGRO_00010166> "forest garden"@en)
+SubClassOf(<http://purl.obolibrary.org/obo/AGRO_00010166> <http://purl.obolibrary.org/obo/AGRO_00000584>)
+
+# Class: <http://purl.obolibrary.org/obo/AGRO_00010167> (shifting cultivation)
```
Comment (celineaubert) [inline src/ontology/agro-edit.owl:5783]: Could you please import the PATO term : http://purl.obolibrary.org/obo/PATO_0000083 ## phase and create the term cropping phase of shifting cultivation, under this PATO term?

File: src/ontology/agro-edit.owl [hunk 6]
```diff
@@ -5446,6 +5462,127 @@ AnnotationAssertion(rdfs:label <http://purl.obolibrary.org/obo/AGRO_00010155> "i
...
+AnnotationAssertion(<http://www.geneontology.org/formats/oboInOwl#hasNarrowSynonym> <http://purl.obolibrary.org/obo/AGRO_00010164> "paddock trees"@en)
+AnnotationAssertion(<http://www.geneontology.org/formats/oboInOwl#hasNarrowSynonym> <http://purl.obolibrary.org/obo/AGRO_00010164> "pasture trees"@en)
+AnnotationAssertion(<http://www.geneontology.org/formats/oboInOwl#hasNarrowSynonym> <http://purl.obolibrary.org/obo/AGRO_00010164> "remnant trees"@en)
+AnnotationAssertion(rdfs:comment <http://purl.obolibrary.org/obo/AGRO_00010164> "Scattered trees may enhance the provision of ecosystem services that might benefit farmers and owners of rural properties, such as pollination of crops, shading for cattle, regulation of nitrogen dynamics and carbon sequestration, herbaceous production, and wood provision."@en)
+AnnotationAssertion(rdfs:label <http://purl.obolibrary.org/obo/AGRO_00010164> "scattered trees"@en)
+SubClassOf(<http://purl.obolibrary.org/obo/AGRO_00010164> <http://purl.obolibrary.org/obo/AGRO_00000585>)
+
+# Class: <http://purl.obolibrary.org/obo/AGRO_00010165> (taungya farming)
```
Comment (celineaubert) [inline src/ontology/agro-edit.owl:5765]: Could you please import the PATO term : http://purl.obolibrary.org/obo/PATO_0000083 ## phase and move the cropping phase of taungya term under this PATO term?

**Comment by celineaubert** [comment]
Hi Krishna, I've shared my comments either by adding comments on the commit or by direclty modifying the terms in Protege. Hope this is not too messy.

File: src/ontology/agro-edit.owl (hunk 5 above)
Comment (KrishnaTO) [reply src/ontology/agro-edit.owl:5783]: Added issue for parent class 'Phase' import (#169)
Updated label and parent class to imported class. 
Commit e62bcd7e4088f6a460ab6e16ed8bb5e1ca08c75b

File: src/ontology/agro-edit.owl (hunk 6 above)
Comment (KrishnaTO) [reply src/ontology/agro-edit.owl:5765]: Added issue for parent class 'Phase' import (#169)
Updated label and parent class to imported class. 
Commit af1d207ba51d4f8b07d0abed9ed89af9c1e10e98

**Pull request closed without merging**